django/db/backends/sqlite3/base.py

```diff
@@ -202,48 +203,51 @@ def get_connection_params(self):
     @async_unsafe
     def get_new_connection(self, conn_params):
         conn = Database.connect(**conn_params)
-        conn.create_function("django_date_extract", 2, _sqlite_datetime_extract)
-        conn.create_function("django_date_trunc", 2, _sqlite_date_trunc)
+        create_deterministic_function = conn.create_function
+        if PY38:
+            create_deterministic_function = functools.partial(conn.create_function, deterministic=True)
+        create_deterministic_function('django_date_extract', 2, _sqlite_datetime_extract)
+        create_deterministic_function('django_date_trunc', 2, _sqlite_date_trunc)
         conn.create_function('django_datetime_cast_date', 3, _sqlite_datetime_cast_date)
         conn.create_function('django_datetime_cast_time', 3, _sqlite_datetime_cast_time)
         conn.create_function('django_datetime_extract', 4, _sqlite_datetime_extract)
         conn.create_function('django_datetime_trunc', 4, _sqlite_datetime_trunc)
```

Comment:
After a quick look I _think_ these can be considered deterministic as well because they don't rely on a connection state directly; they expect `conn_tzname` to be passed as argument.

Reply:
Yes, indeed.